tools/osbuilder/node-builder/azure-linux/README.md

```diff
@@ -0,0 +1,285 @@
+  - Example: `az vm create --resource-group <rg_name> --name <vm_name> --os-disk-size-gb <e.g. 60> --public-ip-sku Standard --size <e.g. Standard_DC4as_cc_v5> --admin-username azureuser --ssh-key-values <ssh_pubkey> --image <MicrosoftCBLMariner:cbl-mariner:...> --security-type Standard`
+- Deploy a [Confidential Containers for AKS cluster](https://learn.microsoft.com/en-us/azure/aks/deploy-confidential-containers-default-policy) via `az aks create`. Note, this way the bits built in this guide will already be present on the cluster's Azure Linux based nodes.
+  - Deploy a debugging pod onto one of the nodes, SSH onto the node.
+- Not validated: Install [Azure Linux](https://github.com/microsoft/azurelinux) on a bare metal machine supporting AMD SEV-SNP.
+
+The following steps assume the user has direct console access on the environnment that was set up.
+
+# Refresh the DNF cache
+```sudo dnf -y makecache```
+
+# Deploy required virtualization packages (e.g., VMM, kernel and Microsoft Hypervisor)
+
+Note: This step can be skipped if your environment was set up through `az aks create`
+
+Install relevant packages:
+```sudo dnf -y install kata-packages-host```
+
+Note: We currently use a [forked version](https://github.com/microsoft/confidential-containers-containerd/tree/tardev-v1.7.7) of `containerd` called `containerd-cc` which is installed as part of the `kata-packages-host` package. This containerd version is based on stock containerd with patches to support the Confidential Containers on AKS use case.
+
+Edit the grub config to boot into the SEV SNP capable kernel upon next reboot:
+```
+boot_uuid=$(sudo grep -o -m 1 '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' /boot/efi/boot/grub2/grub.cfg)
+
+sudo sed -i -e 's@load_env -f \$bootprefix\/mariner.cfg@load_env -f \$bootprefix\/mariner-mshv.cfg\nload_env -f $bootprefix\/mariner.cfg\n@'  /boot/grub2/grub.cfg
+
+sudo sed -i -e 's@menuentry "CBL-Mariner"@menuentry "Dom0" {\n    search --no-floppy --set=root --file /HvLoader.efi\n    chainloader /HvLoader.efi lxhvloader.dll MSHV_ROOT=\\\\Windows MSHV_ENABLE=TRUE MSHV_SCHEDULER_TYPE=ROOT MSHV_X2APIC_POLICY=ENABLE MSHV_SEV_SNP=TRUE MSHV_LOAD_OPTION=INCLUDETRACEMETADATA=1\n    boot\n    search --no-floppy --fs-uuid '"$boot_uuid"' --set=root\n    linux $bootprefix/$mariner_linux_mshv $mariner_cmdline_mshv $systemd_cmdline root=$rootdevice\n    if [ -f $bootprefix/$mariner_initrd_mshv ]; then\n    initrd $bootprefix/$mariner_initrd_mshv\n    fi\n}\n\nmenuentry "CBL-Mariner"@'  /boot/grub2/grub.cfg
+```
+
+Reboot the system:
+
+```sudo reboot now```
+
+# Add Kata handler configuration snippets to containerd configuration
+
+Note: This step can be skipped if your environment was set up through `az aks create`.
+
+An editor like `vim` may need to be installed, for example:
+`sudo dnf -y install vim`
+
+Edit `/etc/containerd/config.toml` to set following configuration:
```

Comment:
Make this a 'tee' command. Iterate over readme to command-ize / script-ize more instructions where applicable